lib/thesis/notifications.ex

```diff
@@ -0,0 +1,163 @@
+defmodule Thesis.Notifications do
+  @moduledoc """
+  Builds a map of Notifications that is passed to the client side. The key should be a
+  representation of the category/type of notification (this is used to determine where and how it's
+  displayed). The value is a list of notification strings.
+
+  Example of final map:
+  %{
+    "page-settings": [],
+    "add-page": [],
+    "import-export-restore": [],
+    "other": []
+  }
+  """
+
+  alias Thesis.Backup
+  import Thesis.Config
+  import Thesis.Utilities
+  import Ecto.Query, only: [limit: 2]
+
+  @allowed_notification_keys [
+    "page-settings",
+    "add-page",
+    "import-export-restore",
+    "other",
+    :page_settings,
+    :add_page,
+    :import_export_restore,
+    :other
+  ]
+
+  def all(page) do
+    %{
+      "page-settings" => [],
+      "add-page" => [],
+      "import-export-restore" => [],
+      "other" => []
+    }
+    |> notifications_from_host_app(notifications())
+    |> notifications_regarding_env(Mix.env)
+    |> notifications_regarding_page(page)
+  end
+
+  def notifications_regarding_page(acc, nil), do: acc
+  def notifications_regarding_page(acc, %_{}) do
+    acc
+  end
```

Comment:
We dont have any just yet, but I want to add a notification that alerts the user that the page title and description are not set (if they are not). We can add more page related notifs here.